app/lib/katello/resources/discovery/container_discovery.rb

```diff
@@ -0,0 +1,123 @@
+module Katello
+  class ContainerDiscovery < RepoDiscovery
+    attr_reader :found, :crawled, :to_follow
+    def initialize(url, crawled = [], found = [], to_follow = [],
+                   upstream_credentials_and_search = {
+                     upstream_username: nil,
+                     upstream_password: nil,
+                     search: '*'
+                   })
+      @uri = uri(url)
+      @upstream_username = upstream_credentials_and_search[:upstream_username].empty? ? nil : upstream_credentials_and_search[:upstream_username]
+      @upstream_password = upstream_credentials_and_search[:upstream_password].empty? ? nil : upstream_credentials_and_search[:upstream_password]
```

Comment:
Like elsewhere, you can use `.presence`
```suggestion
      @upstream_username = upstream_credentials_and_search[:upstream_username].presence
      @upstream_password = upstream_credentials_and_search[:upstream_password].presence
```